modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteInfo.java

```diff
@@ -24,32 +24,45 @@
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.reindex.BulkByScrollTask;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
 import static java.util.Collections.unmodifiableMap;
 import static java.util.Objects.requireNonNull;
+import static org.elasticsearch.common.unit.TimeValue.timeValueMinutes;
 
 public class RemoteInfo implements Writeable {
+    /**
+     * Default socket timeout for requests that don't have one set.
+     */
+    public static final TimeValue DEAULT_SOCKET_TIMEOUT = timeValueMinutes(1);
```

Comment:
I think having 30 secs as a default socket timeout for the REST client is a bit on the high side, but reasonable. We do have the same in our REST tests, and the fact that we needed to change that in our own tests makes me think that maybe the default was too restrictive.